rs/state_manager/src/manifest.rs

```diff
@@ -911,6 +911,17 @@ pub fn compute_manifest(
         None => default_hash_plan(&files, max_chunk_size),
     };
 
+    // Currently, the unverified checkpoint marker file should already be removed by the time we reach this point.
+    // To account for possible changes in the checkpointing order, ensure the marker file is ignored if it already exists and
+    // not accidentally counted when building the file table chunk table.
+    debug_assert!(!files
+        .iter()
+        .any(|FileWithSize(p, _)| *p == checkpoint.unverified_checkpoint_marker()));
+    // Note the condition statement should be adjusted accordingly if in the future we allow computing manifest before removing the marker.
+    if !checkpoint.is_checkpoint_verified() {
+        metrics.manifest_computation_including_marker_critical.inc();
```

Comment:
If we continue here with only a critical error alert, should we also ignore the file for the actual manifest hash?

Reply:
That's a good point. I did not add the logic of ignoring files into the main workflow to avoid unnecessary complexity in normal cases. Ignoring the file here inside the if statement seems a good choice.

Reply:
Added the logic of excluding the marker. I also added a test for manifest computation.

Reply:
Unverified checkpoints are invisible thus state sync would not see, advertise or serve such a checkpoint. 

Taking a step back, suppose somehow state sync happens on such a checkpoint, the receiving node would still be able to reconstruct the checkpoint because the manifest is correct. A extra file in the sending node's checkpoint does not affect the receiving side because it only requests chunks based on the manifest.

Reply:
I have changed the code in two places:

In `handle_compute_manifest_request()`, replica crashes if the checkpoint is unverified. This serves as the safeguard. 

The lower-level function `compute_manifest()` could be used in tests and external tools, potentially on unverified checkpoints. It ignores the marker file for the manifest computation.